### PR TITLE
a rather large mkdocs driven refactor

### PR DESCRIPTION
**Pull request by sulrich**
a rather large refactoring

this consolidates the following items: 
- removes unused/unreferenced documents from the repo (this was historic cruft)
- moves sample code in the examples hierarchy into a new portion of the repo
  (src) such that it can be referenced and updated. however, as an interim
  measure, the contents of the sample code has been pulled into the markdown
  files referencing it.
- refactors the configuration section to treat OC/NETCONF/RESTCONF as directly
  addressable topics. adds a certificate authentication section (new)
- removes the TOC from pages that had it. (this is automatically generated by
  mkdocs)
- adds a nav section to the mkdocs.yml file to only reference the docs sections
  with content.

### CODE REVIEW

File: docs/examples/ansible/index.md [hunk 1]
```diff
@@ -5,15 +5,10 @@ date: 2021-03-18 12:17:00 --0600
 categories:
 ---
 
-- [Ansible collections for](#overview)
```
Comment (sulrich) [inline docs/examples/ansible/index.md:8]: TOCs are now generated by mkdocs.  this eliminates the need to generate these internal to the document.

**Review by sulrich — COMMENTED** [review]
i broke the configuration document up into 5 documents. 
- openconfig
- certificate auth (net new)
- RESTCONF
- NETCONF
- security

the openconfig elements which were previously interrupted by the NET/RESTCONF elements are now consolidated and read a little more cohesively.  the security document is effectively just the service acl elements and can be referenced more consistently.

File: docs/configuration/security.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+<paste the content of the default CP from the file created>
+(config)# 280 permit tcp any any eq 5900
+```
+
+7\. Apply the new ACL
+
+Default VRF
+
+```text
+sysem control-plane
```
Comment (fredhsu) [inline docs/configuration/security.md:118]: typo on 'system control-plane'

File: docs/configuration/security.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+   ip access-group custom-cp in
+```
+
+Non-default VRF
+
+```text
+sysem control-plane
```
Comment (fredhsu) [inline docs/configuration/security.md:125]: typo on 'system control-plane'

**Review by fredhsu — CHANGES_REQUESTED** [review]
Found a couple minor typos, otherwise LGTM, thanks for all the cleanup! @sulrich I rendered on my local machine and it all looks good

File: docs/configuration/security.md (hunk 2 above)
Comment (sulrich) [reply docs/configuration/security.md:118]: ack - thanks!

File: docs/configuration/security.md (hunk 3 above)
Comment (sulrich) [reply docs/configuration/security.md:125]: ack - thanks!

**Review by fredhsu — APPROVED** [review]

File: docs/configuration/openconfig.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+Octa:               No
+Enabled:            Yes
+Server:             running on port 6030, in default VRF
+SSL Profile:        none
+QoS DSCP:           none
+```
+
+### OCTA
+
+The OpenConfing agent (gNMI API) can leverage the EOS state streaming agent's (TerminAttr) libraries, thus exposing EOS
```
[the remark this answers is not in the document]
Comment (sulrich) [reply docs/configuration/openconfig.md]: got it. thanks!

**Pull request merged**